src/mysql_conn.erl

```diff
@@ -667,6 +676,27 @@ log_warnings(#state{socket = Socket, sockmod = SockMod}, Query) ->
              || [Level, Code, Message] <- Rows],
     error_logger:warning_msg("~s in ~s~n", [Lines, Query]).
 
+%% @doc Logs slow queries. Query is the query that gave the warnings.
+maybe_log_slow_query(#state{log_slow_queries = true}, S, Type, N, Query)
+  when S band ?SERVER_QUERY_WAS_SLOW /= 0 ->
+    IndexHint = if
+        S band ?SERVER_STATUS_NO_GOOD_INDEX_USED /= 0 ->
+            " (with no good index)";
+        S band ?SERVER_STATUS_NO_INDEX_USED /= 0 ->
+            " (with no index)";
+        true ->
+            ""
+    end,
+    QueryNumHint = if
+        Type =:= multi_result ->
```

Comment:
A typical case for case.
```Erlang
QueryNumHint =
    case Type of
        multi_result ->
            ...;
        single_result ->
            ...
    end,
```

Reply:
Replaced with a case, though it's a little different now because of the altogether removal of `Type`.